### PR TITLE
Cucumber Hooks

### PR DESCRIPTION
**Pull request by sriteja777**
## Proposed changes
Related to https://github.com/webdriverio/webdriverio/issues/10894 and https://github.com/webdriverio/webdriverio/issues/10879

1. Removed the wrapping for steps. Wrapping only in case of hooks.
2. Store and send hook details (name, tags, type and body).

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by sriteja777** [comment]
@christian-bromann need some validation around for the changes and format followed. Please have a hook in your free time.
Will add more tests and fix tests after the changes looks good to you.

File: packages/wdio-cucumber-framework/src/index.ts [hunk 1]
```diff
@@ -483,6 +487,17 @@ class CucumberAdapter {
...
+            let stepArg = hookParams?.step
+            if (!isStep) {
+                stepArg = {
+                    ...stepArg,
+                    name: options._wdioOptions?.name,
+                    tags: options._wdioOptions?.tags,
+                    hookType: options._wdioOptions?.hookType,
+                    body: options._wdioOptions?.body
+                }
+            }
```
Comment (christian-bromann) [inline packages/wdio-cucumber-framework/src/index.ts:499]: This seems to be more clear
```suggestion
            const stepArg = isStep
                ? hookParams?.step
                : {
                    name: options._wdioOptions?.name,
                    tags: options._wdioOptions?.tags,
                    hookType: options._wdioOptions?.hookType,
                    body: options._wdioOptions?.body
                }
```

File: packages/wdio-cucumber-framework/src/types.ts [hunk 2]
```diff
@@ -133,8 +133,16 @@ export interface HookParams {
...
+    name?: string,
+    tags?: string[],
+    hookType?: string,
+    body: string
+}
+
 export interface StepDefinitionOptions {
-    retry: number
+    retry: number,
+    _wdioOptions?: HookOptions
```
Comment (christian-bromann) [inline packages/wdio-cucumber-framework/src/types.ts:145]: What is the parameter suppose to do? Is it just internally used or also for users?

**Review by christian-bromann — COMMENTED** [review]
Did some initial code review. I will have to look into this change from end to end once I get the build fixed. This will also tell us if we don't see any regressions.

**Comment by christian-bromann** [comment]
@sriteja777 it seems like some tests are failing:

```
 ❯ packages/wdio-cucumber-framework/tests/adapter.test.ts:308:34
    306|         expect(Cucumber.setDefinitionFunctionWrapper).toBeCalledTimes(…
    307|         vi.mocked(Cucumber.setDefinitionFunctionWrapper).mock.calls[0]…
    308|         expect(adapter.wrapStep).toBeCalledWith(
       |                                  ^
    309|             expect.any(Function),
    310|             true,
```

**Comment by christian-bromann** [comment]
@sriteja777 are you still interested in this change? We have updated our Cucumber integration which causes these conflicts.

**Comment by sriteja777** [comment]
Sorry @christian-bromann, didn't get time to check on this. Will check the new cucumber integration and get back to you, if that solves this issue, then this PR can be closed

**Comment by christian-bromann** [comment]
@sriteja777 any updates on this?

**Comment by christian-bromann** [comment]
@sriteja777 I will go ahead and close the PR since more conflicts will crop up .. please re-raise it when ready. Thanks!

**Pull request closed without merging**